### PR TITLE
task(SDK-3620) - Upgrades AGP to 8.2.1

### PR DESCRIPTION
**Pull request by Anush-Shand**
https://wizrocket.atlassian.net/browse/SDK-3620

Upgrades to AGP 8 and addresses build issues because of missing classes when minify is enabled

### CODE REVIEW

**Comment by Anush-Shand** [comment]
The missing classes error for exoplayer and glide has been addressed as they pertain to optional dependencies in the clevertap-sdk. 
For Huawei and Baidu these errors arise from their dependency. Their docs respectively specify adding the  produard  -dontwarn rules 
Do we wish to add rules for these 3rd party libraries to our consumer-rules itself or leave this to the client. Adding it to our consumer-rules may mask some other issue for the client

File: clevertap-core/build.gradle [hunk 1]
```diff
@@ -28,6 +28,8 @@ android {
     kotlinOptions {
         jvmTarget = JavaVersion.VERSION_1_8
     }
+    namespace 'com.clevertap.android.sdk'
+    testNamespace 'com.clevertap.demo'
```
Comment (CTLalit) [inline clevertap-core/build.gradle:32]: not sure why is this change needed

File: gradle-scripts/checkstyle.gradle [hunk 2]
```diff
@@ -17,10 +17,10 @@ task Checkstyle(type: Checkstyle) {
     showViolations true
     include '**/*.java'
     classpath = files()
-    reports{
-        html.enabled(true)
-        html.destination = file("${buildDir}/reports/checkstyle/checkstyle-${project.name}.html")
-        xml.enabled(false)
+    reports {
+        html.required.set(true)
+        xml.required.set(false)
```
Comment (CTLalit) [inline gradle-scripts/checkstyle.gradle:22]: is enabled changed to required?

File: gradle.properties [hunk 3]
```diff
@@ -1,2 +1,5 @@
+android.defaults.buildfeatures.buildconfig=true
+android.nonFinalResIds=false
+android.nonTransitiveRClass=false
 android.useAndroidX=true
-org.gradle.jvmargs=-Xmx4g -XX:MaxPermSize=2048m
\ No newline at end of file
+org.gradle.jvmargs=-Xmx4g
```
Comment (CTLalit) [inline gradle.properties:5]: why removed the RAM config?

File: sample/build.gradle [hunk 4]
```diff
@@ -142,7 +142,7 @@ dependencies {
     remoteImplementation("com.clevertap.android:clevertap-geofence-sdk:1.2.0")
     remoteImplementation("com.clevertap.android:clevertap-xiaomi-sdk:1.5.2")
     remoteImplementation("com.clevertap.android:push-templates:1.1.0")
-    remoteImplementation("com.clevertap.android:clevertap-hms-sdk:1.3.2")
+    remoteImplementation("com.clevertap.android:clevertap-hms-sdk:1.3.3")
 
     stagingImplementation("com.clevertap.android:clevertap-android-sdk:5.2.1")
```
Comment (CTLalit) [inline sample/build.gradle:147]: upgrade this as well?

File: instantapp/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -1,6 +1,5 @@
 <?xml version="1.0" encoding="utf-8"?>
 <manifest xmlns:android="http://schemas.android.com/apk/res/android"
-    package="com.clevertap.demo"
```
Comment (CTLalit) [inline instantapp/src/main/AndroidManifest.xml:3]: why is this removed, does it work properly without it?

File: clevertap-core/build.gradle (hunk 1 above)
Comment (Anush-Shand) [reply clevertap-core/build.gradle:32]: https://developer.android.com/build/releases/past-releases/agp-8-0-0-release-notes#namespace-dsl
Namespace is now needed in the gradle file and not the manifest file

File: gradle-scripts/checkstyle.gradle (hunk 2 above)
Comment (Anush-Shand) [reply gradle-scripts/checkstyle.gradle:22]: This option was removed in JDK 17

File: gradle.properties (hunk 3 above)
Comment (Anush-Shand) [reply gradle.properties:5]: This options was removed in JDK 17

File: sample/build.gradle (hunk 4 above)
Comment (Anush-Shand) [reply sample/build.gradle:147]: Yess. All the sample app changes can be updated in a separated pr? I shall revert this for now

File: instantapp/src/main/AndroidManifest.xml (hunk 5 above)
Comment (Anush-Shand) [reply instantapp/src/main/AndroidManifest.xml:3]: Yess. Namespace is now needed in the gradle file and not the manifest file

File: gradle-scripts/checkstyle.gradle (hunk 2 above)
Comment (Anush-Shand) [reply gradle-scripts/checkstyle.gradle:22]: Yess

File: clevertap-core/build.gradle (hunk 1 above)
Comment (Anush-Shand) [reply clevertap-core/build.gradle:32]: The testNamespace is picked up from the AndroidManifest of androidTest of the core module

**Review by CTLalit — APPROVED** [review]
